### ISSUE TITLE
descriptions for codes

### ISSUE THREAD

**Issue by pvdbosch** [belgif/rest-guide#101]
Formulate guidelines when to include descriptions of codes in each message, or only in refData resources.
Include href or link to code description via mapping to a URI via JSON-LD?

Guidelines could allow some options depending on clients that an API is targeting, e.g. backend-only (business data), or for a user interface (Backend For Frontend API).

**Comment by pvdbosch** [comment]
eHealth: no hrefs currently
Smals: inconsistent between APIs, only hrefs within single API

**Comment by JDMKSZ** [comment]
It seems to me that adding hrefs only it the most RESTful? See [this SO post](https://stackoverflow.com/questions/30981727/link-to-another-resource-in-a-rest-api-by-its-id-or-by-its-url) + the link to the Fielding article

**Comment by pvdbosch** [comment]
Using only hyperlinks to look up and navigate between resources (HATEOS) is indeed part of the original REST principles of Roy Fielding, but in practice doesn't seems feasible with current technological state of REST APIs.
Some more background on this can be found [here in the zalando rest guide](http://opensource.zalando.com/restful-api-guidelines/#163). REST clients tend to use URI templates or be hardcoded in another way to the API's URI space (i.e. with code generation from OpenAPI).

Using the href links from responses might be more common for JavaScript clients, but I think it's overall less used then constructing the URI from a URI template (which may be generated from OpenAPI).

I'd propose to amend to guide to limit the use of "href" linking (in combanation with the resource identifier) only for resources within the same API to ensure low coupling between different APIs (e.g. no impact when there's new major version of an API).

**Comment by pvdbosch** [comment]
agreed in WG to specify that "href" should only be used to refer within the same API.
Further rules are difficult to generalize for every use case.
